src/Sarif.Sarifer/BackgroundAnalysisService.cs

```diff
@@ -49,5 +47,18 @@ public async Task AnalyzeAsync(string logId, IEnumerable<string> targetFiles, Ca
 
             await Task.WhenAll(tasks).ConfigureAwait(continueOnCapturedContext: false);
         }
+
+        /// <inheritdoc/>
+        public async Task ClearResultsAsync()
+        {
+            var tasks = new List<Task>(this.analyzers.Count());
+
+            foreach (IBackgroundAnalyzer analyzer in this.analyzers)
+            {
+                tasks.Add(analyzer.ClearResultsAsync());
```

Comment:
>ClearResultsAsync [](start = 35, length = 17)

My first design instinct is that analyzers should not be responsible for clearing results. Analyzers should be responsible for one thing: analyzing. My instinct is that the background analysis service should keep track of the results and be responsible for clearing them. But I will read further and I might change my mind.

Reply:
yeah...its strange for me as well...the issue in this case is the sink, which is available in the `BackgroundAnalyzerBase.cs`. Based on that, what we would have to do is: analyze propagate the results to the Service and the Service would be responsible to send to sink, clean, etc. And, then, we will have some issues with the dispose of the stream. Will have to check how to work on that, since we will have to propagate it.

---
In reply to: [535424123](https://github.com/microsoft/sarif-visualstudio-extension/pull/310#discussion_r535424123) [](ancestors = 535424123)

Reply:
Yes, you should remove these from the analyzer interface and implementation. The goal of this code was not to enable aggregation of multiple background analyzers, so let's be careful about building this out further. Today, it should be sufficient that the analyzers write to the SARIF log. i.e., they shouldn't retain any state/etc. that would require them to get a notification of close/clear.


---
In reply to: [535424123](https://github.com/microsoft/sarif-visualstudio-extension/pull/310#discussion_r535424123) [](ancestors = 535424123)

Reply:
Moved the sink to the service class. Let me know what do you think!
Thank you

---
In reply to: [535442762](https://github.com/microsoft/sarif-visualstudio-extension/pull/310#discussion_r535442762) [](ancestors = 535442762,535424123)